xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp

```diff
@@ -144,7 +144,7 @@ static void LoadTexture(GLenum target
   *v = (GLfloat)height / height2;
 }
 
-COverlayTextureGL::COverlayTextureGL(CDVDOverlayImage* o)
+COverlayTextureGL::COverlayTextureGL(CDVDOverlayImage* o, int sourceWidth, int sourceHeight)
```

Comment:
it gets worse. CDVDOverlayImage does already know sourceWidth and CRect of sub image. further you can't allign to video if pos is set to POSITION_RELATIVE. that would break rendering in video control.

see my post on the issue where code needs to be changed.

Reply:
CDVDOverlayImage know sourceWidth of sub, and the parament is sourceWidth of video

Reply:
CDVDOverlayImage knows source of video, which is source_width. width/height if subtitle is in width/height of CDVDOverlayImage.

Reply:
if video size is 1920x800 and sub size is 1920x1080, then sourceHeight is 800 and source_height is 1080 here.

Reply:
I think this is always the same. If you don't think so, please provide a sample.

Reply:
thanks. ok, then passing videoSourceWidth here is fine. if subSource != videoSource, we need to rescale the subs from subSource to videoSource and set alignment to ALIGN_VIDEO for all cases.